packages/manager/apps/telecom/src/app/telecom/pack/slots/hostedEmail/detail/detail.controller.js

```diff
@@ -0,0 +1,241 @@
+import assignIn from 'lodash/assignIn';
+import filter from 'lodash/filter';
+import isObject from 'lodash/isObject';
+import map from 'lodash/map';
+import snakeCase from 'lodash/snakeCase';
+
+export default class PackHostedEmailDetailCtrl {
+  /* @ngInject */
+  constructor(
+    $q,
+    $state,
+    $http,
+    $translate,
+    TucToast,
+    OvhApiPackXdsl,
+    OvhApiPackXdslHostedEmail,
+  ) {
+    this.$q = $q;
+    this.$state = $state;
+    this.$http = $http;
+    this.$translate = $translate;
+    this.TucToast = TucToast;
+    this.OvhApiPackXdsl = OvhApiPackXdsl;
+    this.OvhApiPackXdslHostedEmail = OvhApiPackXdslHostedEmail;
+  }
+
+  $onInit() {
+    this.askForDelete = false;
+    this.accountDeleting = false;
+    this.loading = true;
+
+    this.passwordRules = [
+      {
+        id: 'length',
+        caption: this.$translate.instant(
+          'hosted_email_detail_change_password_rule_size',
+        ),
+        validator(str) {
+          return str && str.length > 7 && str.length < 21;
+        },
+      },
+      {
+        id: 'specialChar',
+        caption: this.$translate.instant(
+          'hosted_email_detail_change_password_rule_special',
+          { list: '#{}()[]-|@=*+/!:;' },
+        ),
+        validator: /^[\w~"#'{}(\\)[\]\-|\\^@=*+/!:;.,?<>%*µÀÁÂÃÄÅàáâãäåÒÓÔÕÖØòóôõöøÈÉÊËèéêëÇçÌÍÎÏìíîïÙÚÛÜùúûüÿÑñ]+$/,
+        immediateWarning: true,
+      },
+    ];
+
+    return this.$q
+      .all([this.loadConfiguration(), this.getAccount(), this.loadPackInfo()])
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  toHuman(valueParam) {
+    let value = valueParam;
+    const orig = value;
+    const units = [
+      'hosted_email_detail_byte',
+      'hosted_email_detail_Kb',
+      'hosted_email_detail_Mb',
+      'hosted_email_detail_Gb',
+      'hosted_email_detail_Tb',
+    ];
+    let index = 0;
+    while (value / 1024 > 1 && index < units.length) {
+      value /= 1024;
+      index += 1;
+    }
```

Comment:
In fact, you're converting the value to some unit which quantity is lower than 1024? This should be, at least refactored in a separate method, as it is hard to understand directly what is done here